compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -3930,8 +3930,9 @@ public void visit(BLangTernaryExpr ternaryExpr) {
             } else if (types.isAssignable(thenType, elseType)) {
                 resultType = elseType;
             } else {
-                dlog.error(ternaryExpr.pos, DiagnosticErrorCode.INCOMPATIBLE_TYPES, thenType, elseType);
-                resultType = symTable.semanticError;
+                resultType = BUnionType.create(null, elseType, thenType);
```

Comment:
Shall we do 
```suggestion
                resultType = BUnionType.create(null, thenType, elseType);
```

Just so that the type reads in the same order as the expressions.

Reply:
Will update.